classes/task/FileLoader.php

```diff
@@ -28,13 +28,15 @@
 abstract class FileLoader extends ScheduledTask
```

Comment:
Imagine that this task ran, and created many jobs to process the files.
Then, while the jobs are still being processed (e.g. maybe there are so many that it's taking days to finish), this task gets executed again.

Is it going to generate issues (reprocessing)? Is it worth to give IDs to the jobs to avoid them from being duplicated on the queue?

Reply:
There are two cases that would need attention:
- If the re-processing of the same log file has started next day: This case can only occur if the admin explicitly does it. But this we could eventually solve by having an ID, to avoid them from being duplicated on the queue. I will need to think about how to solve it. Maybe in a new issue. It is probably also a rare case, because I hope that admin will not do such a think :-)
- If log files processing (e.g. when the admin re-process the whole month) takes longer than a day, and the normal scheduled task that runs every day (and that would process the log file from yesterday) is run on the next day: Then the new scheduled task would create a chain of the new jobs (for the new log file) and those can be run in parallel with the jobs from the re-processing the whole month. This would again lead to enormous DB tables that would break the processing. This I do not know how to solve, I would need to investigate what is possible. But also this I would like to leave for a new issue.

Reply:
I've skimmed through the documentation of Laravel queue and I remember there were some features to address such problems. I'm just trying to help with questions, feel free to create new issues for the things that make sense :)

Reply:
@bozana @jonasraoni I think what we are looking here in this case is making the jobs unique . It should be very easy just by making any job class implementing `Illuminate\Contracts\Queue\ShouldBeUnique` . Need to bit checking to make sure it can properly interact with the cache but seems ok to me and no changes require . see more at https://laravel.com/docs/10.x/queues#unique-jobs .